{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "racial-world",
   "metadata": {},
   "source": [
    "# Processing of GWAS summary statistics\n",
    "## Previous attempt: GWAS Curation\n",
    "Analyzing just three traits have demonstrated differences in either onset or incidence between males and females has not revealed any obvious differences in heritability explained by marginal and sex-specific mQTLs. As such, I've decided to grab traits from high level categories in the [EFO ontology](https://www.ebi.ac.uk/gwas/docs/ontologyhttps://www.ebi.ac.uk/gwas/docs/ontology) over GWAS in the EBI GWAS catalogue. I am focusing primarily on neurobehavioral traits, citing the following study which found that the majority of traits do not show sex-dependent genetic effects ([Stringer et al 2017](https://doi-org.ezproxy.library.ubc.ca/10.1038/s41598-017-09249-3))."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "configured-fleet",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The rpy2.ipython extension is already loaded. To reload it, use:\n",
      "  %reload_ext rpy2.ipython\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "\n",
       "        <span>StrVector with 17 elements.</span>\n",
       "        <table>\n",
       "        <tbody>\n",
       "          <tr>\n",
       "          \n",
       "            <td>\n",
       "            'forcats'\n",
       "            </td>\n",
       "          \n",
       "            <td>\n",
       "            'stringr'\n",
       "            </td>\n",
       "          \n",
       "            <td>\n",
       "            'dplyr'\n",
       "            </td>\n",
       "          \n",
       "            <td>\n",
       "            ...\n",
       "            </td>\n",
       "          \n",
       "            <td>\n",
       "            'datasets'\n",
       "            </td>\n",
       "          \n",
       "            <td>\n",
       "            'methods'\n",
       "            </td>\n",
       "          \n",
       "            <td>\n",
       "            'base'\n",
       "            </td>\n",
       "          \n",
       "          </tr>\n",
       "        </tbody>\n",
       "        </table>\n",
       "        "
      ],
      "text/plain": [
       "<rpy2.robjects.vectors.StrVector object at 0x2aaaae35d4c0> [RTYPES.STRSXP]\n",
       "R classes: ('character',)\n",
       "['forcats', 'stringr', 'dplyr', 'purrr', ..., 'utils', 'datasets', 'methods', 'base']"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "%load_ext rpy2.ipython\n",
    "%R library(tidyverse)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "macro-jesus",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DISEASE/TRAIT</th>\n",
       "      <th>STUDY ACCESSION</th>\n",
       "      <th>DATE ADDED TO CATALOG</th>\n",
       "      <th>PUBMED ID</th>\n",
       "      <th>FIRST AUTHOR.x</th>\n",
       "      <th>DATE.x</th>\n",
       "      <th>JOURNAL</th>\n",
       "      <th>LINK</th>\n",
       "      <th>STUDY</th>\n",
       "      <th>INITIAL SAMPLE SIZE</th>\n",
       "      <th>...</th>\n",
       "      <th>STAGE</th>\n",
       "      <th>NUMBER OF INDIVDUALS</th>\n",
       "      <th>BROAD ANCESTRAL CATEGORY</th>\n",
       "      <th>COUNTRY OF ORIGIN</th>\n",
       "      <th>COUNTRY OF RECRUITMENT</th>\n",
       "      <th>ADDITONAL ANCESTRY DESCRIPTION</th>\n",
       "      <th>EFO term</th>\n",
       "      <th>EFO URI</th>\n",
       "      <th>Parent term</th>\n",
       "      <th>Parent URI</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>&amp;beta;2-Glycoprotein I (&amp;beta;2-GPI) plasma le...</td>\n",
       "      <td>GCST001800</td>\n",
       "      <td>2013-03-19</td>\n",
       "      <td>23279374</td>\n",
       "      <td>Athanasiadis G</td>\n",
       "      <td>2013-01-02</td>\n",
       "      <td>J Thromb Haemost</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/23279374</td>\n",
       "      <td>Genetic determinants of plasma β₂-glycoprotein...</td>\n",
       "      <td>306 European ancestry individuals</td>\n",
       "      <td>...</td>\n",
       "      <td>initial</td>\n",
       "      <td>306.0</td>\n",
       "      <td>European</td>\n",
       "      <td>NR</td>\n",
       "      <td>Spain</td>\n",
       "      <td>NaN</td>\n",
       "      <td>glycoprotein measurement</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004555</td>\n",
       "      <td>Other measurement</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0001444</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1,5-anhydroglucitol levels</td>\n",
       "      <td>GCST004643</td>\n",
       "      <td>2017-09-05</td>\n",
       "      <td>28588231</td>\n",
       "      <td>Li M</td>\n",
       "      <td>2017-06-06</td>\n",
       "      <td>Sci Rep</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/28588231</td>\n",
       "      <td>Genome-wide association study of 1,5-anhydrogl...</td>\n",
       "      <td>7,550 European ancestry individuals</td>\n",
       "      <td>...</td>\n",
       "      <td>initial</td>\n",
       "      <td>7550.0</td>\n",
       "      <td>European</td>\n",
       "      <td>NR</td>\n",
       "      <td>U.S.</td>\n",
       "      <td>European American</td>\n",
       "      <td>1,5 anhydroglucitol measurement</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0008009</td>\n",
       "      <td>Other measurement</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0001444</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1,5-anhydroglucitol levels</td>\n",
       "      <td>GCST004643</td>\n",
       "      <td>2017-09-05</td>\n",
       "      <td>28588231</td>\n",
       "      <td>Li M</td>\n",
       "      <td>2017-06-06</td>\n",
       "      <td>Sci Rep</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/28588231</td>\n",
       "      <td>Genome-wide association study of 1,5-anhydrogl...</td>\n",
       "      <td>7,550 European ancestry individuals</td>\n",
       "      <td>...</td>\n",
       "      <td>replication</td>\n",
       "      <td>2030.0</td>\n",
       "      <td>African American or Afro-Caribbean</td>\n",
       "      <td>NR</td>\n",
       "      <td>U.S.</td>\n",
       "      <td>African American</td>\n",
       "      <td>1,5 anhydroglucitol measurement</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0008009</td>\n",
       "      <td>Other measurement</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0001444</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1,5-anhydroglucitol levels</td>\n",
       "      <td>GCST004643</td>\n",
       "      <td>2017-09-05</td>\n",
       "      <td>28588231</td>\n",
       "      <td>Li M</td>\n",
       "      <td>2017-06-06</td>\n",
       "      <td>Sci Rep</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/28588231</td>\n",
       "      <td>Genome-wide association study of 1,5-anhydrogl...</td>\n",
       "      <td>7,550 European ancestry individuals</td>\n",
       "      <td>...</td>\n",
       "      <td>replication</td>\n",
       "      <td>8790.0</td>\n",
       "      <td>European</td>\n",
       "      <td>NR</td>\n",
       "      <td>Germany, U.K.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1,5 anhydroglucitol measurement</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0008009</td>\n",
       "      <td>Other measurement</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0001444</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>17-hydroxyprogesterone (17-OHP) levels</td>\n",
       "      <td>GCST008879</td>\n",
       "      <td>2019-10-21</td>\n",
       "      <td>31169883</td>\n",
       "      <td>Pott J</td>\n",
       "      <td>2019-06-06</td>\n",
       "      <td>J Clin Endocrinol Metab</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/31169883</td>\n",
       "      <td>Genetic association study of eight steroid hor...</td>\n",
       "      <td>1,358 European ancestry men, 712 European ance...</td>\n",
       "      <td>...</td>\n",
       "      <td>initial</td>\n",
       "      <td>2070.0</td>\n",
       "      <td>European</td>\n",
       "      <td>NR</td>\n",
       "      <td>Germany</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17-hydroxyprogesterone measurement</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0010220</td>\n",
       "      <td>Other measurement</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0001444</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 37 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       DISEASE/TRAIT STUDY ACCESSION  \\\n",
       "0  &beta;2-Glycoprotein I (&beta;2-GPI) plasma le...      GCST001800   \n",
       "1                         1,5-anhydroglucitol levels      GCST004643   \n",
       "2                         1,5-anhydroglucitol levels      GCST004643   \n",
       "3                         1,5-anhydroglucitol levels      GCST004643   \n",
       "4             17-hydroxyprogesterone (17-OHP) levels      GCST008879   \n",
       "\n",
       "  DATE ADDED TO CATALOG  PUBMED ID  FIRST AUTHOR.x      DATE.x  \\\n",
       "0            2013-03-19   23279374  Athanasiadis G  2013-01-02   \n",
       "1            2017-09-05   28588231            Li M  2017-06-06   \n",
       "2            2017-09-05   28588231            Li M  2017-06-06   \n",
       "3            2017-09-05   28588231            Li M  2017-06-06   \n",
       "4            2019-10-21   31169883          Pott J  2019-06-06   \n",
       "\n",
       "                   JOURNAL                                  LINK  \\\n",
       "0         J Thromb Haemost  www.ncbi.nlm.nih.gov/pubmed/23279374   \n",
       "1                  Sci Rep  www.ncbi.nlm.nih.gov/pubmed/28588231   \n",
       "2                  Sci Rep  www.ncbi.nlm.nih.gov/pubmed/28588231   \n",
       "3                  Sci Rep  www.ncbi.nlm.nih.gov/pubmed/28588231   \n",
       "4  J Clin Endocrinol Metab  www.ncbi.nlm.nih.gov/pubmed/31169883   \n",
       "\n",
       "                                               STUDY  \\\n",
       "0  Genetic determinants of plasma β₂-glycoprotein...   \n",
       "1  Genome-wide association study of 1,5-anhydrogl...   \n",
       "2  Genome-wide association study of 1,5-anhydrogl...   \n",
       "3  Genome-wide association study of 1,5-anhydrogl...   \n",
       "4  Genetic association study of eight steroid hor...   \n",
       "\n",
       "                                 INITIAL SAMPLE SIZE  ...        STAGE  \\\n",
       "0                  306 European ancestry individuals  ...      initial   \n",
       "1                7,550 European ancestry individuals  ...      initial   \n",
       "2                7,550 European ancestry individuals  ...  replication   \n",
       "3                7,550 European ancestry individuals  ...  replication   \n",
       "4  1,358 European ancestry men, 712 European ance...  ...      initial   \n",
       "\n",
       "  NUMBER OF INDIVDUALS            BROAD ANCESTRAL CATEGORY COUNTRY OF ORIGIN  \\\n",
       "0                306.0                            European                NR   \n",
       "1               7550.0                            European                NR   \n",
       "2               2030.0  African American or Afro-Caribbean                NR   \n",
       "3               8790.0                            European                NR   \n",
       "4               2070.0                            European                NR   \n",
       "\n",
       "  COUNTRY OF RECRUITMENT ADDITONAL ANCESTRY DESCRIPTION  \\\n",
       "0                  Spain                            NaN   \n",
       "1                   U.S.              European American   \n",
       "2                   U.S.               African American   \n",
       "3          Germany, U.K.                            NaN   \n",
       "4                Germany                            NaN   \n",
       "\n",
       "                             EFO term                               EFO URI  \\\n",
       "0            glycoprotein measurement  http://www.ebi.ac.uk/efo/EFO_0004555   \n",
       "1     1,5 anhydroglucitol measurement  http://www.ebi.ac.uk/efo/EFO_0008009   \n",
       "2     1,5 anhydroglucitol measurement  http://www.ebi.ac.uk/efo/EFO_0008009   \n",
       "3     1,5 anhydroglucitol measurement  http://www.ebi.ac.uk/efo/EFO_0008009   \n",
       "4  17-hydroxyprogesterone measurement  http://www.ebi.ac.uk/efo/EFO_0010220   \n",
       "\n",
       "         Parent term                            Parent URI  \n",
       "0  Other measurement  http://www.ebi.ac.uk/efo/EFO_0001444  \n",
       "1  Other measurement  http://www.ebi.ac.uk/efo/EFO_0001444  \n",
       "2  Other measurement  http://www.ebi.ac.uk/efo/EFO_0001444  \n",
       "3  Other measurement  http://www.ebi.ac.uk/efo/EFO_0001444  \n",
       "4  Other measurement  http://www.ebi.ac.uk/efo/EFO_0001444  \n",
       "\n",
       "[5 rows x 37 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Loading data from the GWAS catalogue\n",
    "gwas_meta = pd.read_csv(\"../data/gwas_catalog_all.csv\")\n",
    "gwas_meta.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "everyday-circuit",
   "metadata": {},
   "source": [
    "## Uniform formatting with `munge_sumstats.py`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "copyrighted-height",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../../tmp_GWAS/pgc_sumstats/adhd_jul2017.gz\n",
      "CHR\tSNP\tBP\tA1\tA2\tINFO\tOR\tSE\tP\n",
      "1\trs202152658\t751343\tA\tT\t0.884\t1.03118\t0.0221\t0.1654\n",
      "../../tmp_GWAS/pgc_sumstats/anxiety.meta.full.cc.tbl.gz\n",
      "SNPID\tCHR\tBP\tAllele1\tAllele2\tFreq1\tEffect\tStdErr\tP.value\tTotalN\n",
      "rs1000033\t1\t226580387\tt\tg\t0.8266\t-0.0574\t0.0348\t0.09867\t17310\n",
      "../../tmp_GWAS/pgc_sumstats/anxiety.meta.full.fs.tbl.gz\n",
      "SNPID\tCHR\tBP\tAllele1\tAllele2\tFreq1\tEffect\tStdErr\tP.value\tTotalN\n",
      "rs1000033\t1\t226580387\tt\tg\t0.824\t-0.0057\t0.0058\t0.3288\t18186\n",
      "../../tmp_GWAS/pgc_sumstats/AUDIT_UKB_2018_AJP.txt.gz\n",
      "chr rsid a_0 a_1 info beta_T se_T p_T beta_C se_C p_C beta_P se_P p_P N\n",
      "1 1:10000179_AAAAAAAC_A AAAAAAAC A 0.985768 -0.0023263 0.0087656 0.790715040934701 0.0024046 0.0080078 0.763958909282397 -0.013933 0.0076296 0.0678265926015363 121568\n",
      "../../tmp_GWAS/pgc_sumstats/Cannabis_ICC_23andmetop_UKB_het.txt.gz\n",
      "CHR\tSNP\tBP\tA1\tA2\tFRQ\tBETA\tSE\tZ\tP\tDirection\tHetISq\tHetDf\tHetPVa\tNca\tNco\tNeff\n",
      "3\trs2875907\t85518580\ta\tg\t0.3524\t0.0712\t0.0086\t8.27907\t9.381e-17\t+++\t0\t2\t0.5839\t53179\t131586\t151493\n",
      "../../tmp_GWAS/pgc_sumstats/iPSYCH-PGC_ASD_Nov2017.gz\n",
      "CHR\tSNP\tBP\tA1\tA2\tINFO\tOR\tSE\tP\n",
      "8\trs62513865\t101592213\tT\tC\t0.949\t1.00652\t0.027\t0.8086\n",
      "../../tmp_GWAS/pgc_sumstats/ocd_aug2017.gz\n",
      "CHR\tSNP\tBP\tA1\tA2\tINFO\tOR\tSE\tP\n",
      "1\trs141242758\t734349\tT\tC\t0.6683\t1.07358\t0.1088\t0.5144\n",
      "../../tmp_GWAS/pgc_sumstats/PGC3_SCZ_wave3_public.v2.tsv.gz\n",
      "CHR\tSNP\tBP\tA1\tA2\tFRQ_A_67390\tFRQ_U_94015\tINFO\tOR\tSE\tP\tngt\tDirection\tHetISqt\tHetDf\tHetPVa\tNca\tNco\tNeff\n",
      "8\trs62513865\t101592213\tC\tT\t0.932\t0.928\t0.954\t1.01349\t0.0171\t0.4319\t0\t-?-+--+++???-+--+-+-++-++-+?????+++-+--+-??++-----++--------++++++-+-?+-+-++---++--+--++++\t9.0\t77\t0.259\t55085\t78957\t60448.13\n",
      "../../tmp_GWAS/pgc_sumstats/pgc_adhd_females.gz\n",
      "SNP CHR BP A1 A2 Effect SE P\n",
      "rs2326918 6 130840091 A G -0.0472 0.0353 0.1818\n",
      "../../tmp_GWAS/pgc_sumstats/pgc_adhd_males.gz\n",
      "SNP CHR BP A1 A2 Effect SE P\n",
      "rs2326918 6 130840091 A G -0.0103 0.0245 0.6731\n",
      "../../tmp_GWAS/pgc_sumstats/pgc_alcdep.afr_discovery.aug2018_release.txt.gz\n",
      "CHR SNP BP A1 A2 Z P Weight\n",
      "1 rs111906919:16160196:T:C 16160196 T C -0.757 0.4488 1978.10\n",
      "../../tmp_GWAS/pgc_sumstats/pgc_alcdep.afr_unrelated.aug2018_release.txt.gz\n",
      "CHR SNP BP A1 A2 Z P Weight\n",
      "1 rs111906919:16160196:T:C 16160196 T C -0.757 0.4488 1978.10\n",
      "../../tmp_GWAS/pgc_sumstats/pgc_alcdep.afr_unrel_genotyped.aug2018_release.txt.gz\n",
      "CHR SNP BP A1 A2 INFO OR SE P Neff\n",
      "8 rs73696507 105231413 C T 0.979 1.02798 0.0731 0.7062 4587.06\n",
      "../../tmp_GWAS/pgc_sumstats/pgc_alcdep.discovery.aug2018_release.txt.gz\n",
      "CHR SNP BP A1 A2 Z P Weight\n",
      "1 rs370839278:194108898:ACAAGGGTG:A 194108898 A ACAAGGGTG 0.833 0.4049 4947.36\n",
      "../../tmp_GWAS/pgc_sumstats/pgc_alcdep.eur_discovery.aug2018_release.txt.gz\n",
      "CHR SNP BP A1 A2 Z P Weight\n",
      "1 rs10799799:23973601:G:T 23973601 T G -0.012 0.9904 24810.32\n",
      "../../tmp_GWAS/pgc_sumstats/pgc_alcdep.eur_unrelated.aug2018_release.txt.gz\n",
      "CHR SNP BP A1 A2 Z P Weight\n",
      "1 rs10799799:23973601:G:T 23973601 T G -0.190 0.8495 21857.28\n",
      "../../tmp_GWAS/pgc_sumstats/pgc_alcdep.eur_unrel_genotyped.aug2018_release.txt.gz\n",
      "CHR SNP BP A1 A2 INFO OR SE P Neff\n",
      "8 rs79643588 106973048 A G 0.995 0.97346 0.0381 0.4797 17852.9\n",
      "../../tmp_GWAS/pgc_sumstats/pgc_alcdep.trans_fe_unrel_geno.aug2018_release.txt.gz\n",
      "CHR SNP BP A1 A2 INFO BETA SE P Neff\n",
      "1 rs12034030 68677760 T C 0.986 0.0242563 0.0270064 0.369097 20293.6\n",
      "../../tmp_GWAS/pgc_sumstats/pgc_alcdep.trans_mantra_unrel_geno.aug2018_release.txt.gz\n",
      "CHR SNP BP A1 A2 INFO Log10BF Neff\n",
      "1 rs12034030 68677760 T C 0.986 -0.42345 20293.6\n",
      "../../tmp_GWAS/pgc_sumstats/pgc_alcdep.trans_re2_unrel_geno.aug2018_release.txt.gz\n",
      "CHR SNP BP A1 A2 INFO STAT1_RE2 STAT2_RE2 P Neff\n",
      "1 rs12034030 68677760 T C 0.986 0.806703 0.00000 0.402449 20293.6\n",
      "../../tmp_GWAS/pgc_sumstats/pgcAN2.2019-07.vcf.tsv.gz\n",
      "##fileFormat=PGCsumstatsVCFv1.0\n",
      "##CAVEAT EMPTOR: ALWAYS CHECK FOR NEWER VERSION\n",
      "../../tmp_GWAS/pgc_sumstats/pgc_bip_2018.gz\n",
      "CHR\tSNP\tBP\tA1\tA2\tFRQ_A_20352\tFRQ_U_31358\tINFO\tOR\tSE\tP\tngt\tDirection\tHetISqt\tHetDf\tHetPVa\tNca\tNco\tNeff\n",
      "10\trs185339560\t2392426\tT\tC\t0.00935\t0.0081\t0.67\t1.16813\t0.0923\t0.09239\t0\t+--+++++--+?++--+-+-?++++++++--+\t0.0\t29\t0.6096\t19294\t29909\t22093.09\n",
      "../../tmp_GWAS/pgc_sumstats/pgc.cross.full.2013-03.txt.gz\n",
      "snpid hg18chr bp a1 a2 or se pval info ngt CEUaf \n",
      "rs3131972\t1\t742584\tA\tG\t1.047\t0.0361\t0.2032\t0.774\t0\t0.16055\n",
      "../../tmp_GWAS/pgc_sumstats/pgc.ed.freeze1.summarystatistics.July2017.txt.gz\n",
      "CHR\tSNP\tBP\tA1\tA2\tINFO\tOR\tSE\tP\tngt\n",
      "1\tchr1:100012367\t100012367\tT\tC\t0.842\t3.84164\t1.5\t0.3694\t0\n",
      "../../tmp_GWAS/pgc_sumstats/PGC_UKB_depression_genome-wide.txt.gz\n",
      "MarkerName A1 A2 Freq LogOR StdErrLogOR P\n",
      "rs2326918 a g 0.8452 0.0106 0.006 0.07561\n",
      "../../tmp_GWAS/pgc_sumstats/pts_all_freeze2_females.results.gz\n",
      "CHR\tSNP\tBP\tA1\tA2\tFRQ_A_16845\tFRQ_U_80105\tINFO\tOR\tSE\tP\tngt\tDirection\tHetISqt\tHetDf\tHetPVa\tNca\tNco\tNeff\n",
      "4\trs34341799\t115156932\tG\tGA\t0.2726\t0.2504\t0.997508\t1.12547\t0.0222\t9.698e-08\t0\t++++-++-+++++-+?-++++?--+++-++-+++++-+++++\t0.0\t39\t0.9923\t9795\t15143\t22188\n",
      "../../tmp_GWAS/pgc_sumstats/pts_all_freeze2_males.results.gz\n",
      "CHR\tSNP\tBP\tA1\tA2\tFRQ_A_12082\tFRQ_U_82978\tINFO\tOR\tSE\tP\tngt\tDirection\tHetISqt\tHetDf\tHetPVa\tNca\tNco\tNeff\n",
      "12\trs71439306\t132944639\tA\tC\t0.0327\t0.0237\t0.753576\t1.54326\t0.0821\t1.273e-07\t0\t+?++-?+++++?++?????--?-+++-+++??????????????-+-+--?\t0.0\t26\t0.7858\t5202\t21351\t14540.4\n",
      "../../tmp_GWAS/pgc_sumstats/pts_all_freeze2_overall.results.gz\n",
      "CHR\tSNP\tBP\tA1\tA2\tFRQ_A_29529\tFRQ_U_166145\tINFO\tOR\tSE\tP\tngt\tDirection\tHetISqt\tHetDf\tHetPVa\tNca\tNco\tNeff\n",
      "6\trs9459288m\t157790612\tT\tC\t0.7144\t0.7312\t0.862271\t0.915212\t0.0170\t1.804e-07\t0\t--+---?-+---++-------+?+-----++-----?++++++---+-+++-+++++-+++++---++--\t15.8\t66\t0.1415\t18424\t48144\t46094.3\n",
      "../../tmp_GWAS/pgc_sumstats/tag.cpd.tbl.gz\n",
      "CHR\tSNP\tBP\tA1\tA2\tFRQ_A\tFRQ_U\tINFO\tOR\tSE\tP\n",
      "1\trs12565286\t711153\tC\tG\t0.0434\t0.0434\t1\t-0.4029\t0.5634\t0.4745\n",
      "../../tmp_GWAS/pgc_sumstats/TS_Oct2018.gz\n",
      "SNP CHR BP A1 A2 INFO OR SE P\n",
      "rs2326918 6 130840091 A G 0.9858 0.965412 0.0349 0.3139\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "for f in ../../tmp_GWAS/pgc_sumstats/*; do\n",
    "    echo $f\n",
    "    zcat $f | head -n 2\n",
    "done\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "accessible-satin",
   "metadata": {},
   "source": [
    "With the exception of the vcf.tsv formatted files, it appears that the formatting of these summary statistics is fairly uniform. Let's see if we can just run munge_sumstats through all of these files:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "standing-church",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process is interrupted.\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "source /home/wcasazza/miniconda3/bin/activate\n",
    "conda activate ldsc\n",
    "cd /scratch/st-dennisjk-1/wcasazza/tmp_GWAS/\n",
    "echo \"\" > missed_files.txt\n",
    "for f in pgc_sumstats/*.gz; do\n",
    "    out_f=${f%.*}\n",
    "    if [[ ! -f \"pgc_formatted_sumstats/${out_f##*/}.sumstats.gz\" ]]; then\n",
    "        cmd=\"/arc/project/st-dennisjk-1/software/ldsc/munge_sumstats.py\"\n",
    "        cmd+=\" --sumstats ${f}\"\n",
    "        cmd+=\" --out pgc_formatted_sumstats/${out_f##*/}\"\n",
    "        for arg in $(zgrep -v \"^#\" ${f} | head -n 1 | tr -d '\\r\\n'); do\n",
    "            case ${arg} in\n",
    "                \"TotalN\")\n",
    "                    cmd+=\" --N-col TotalN\"\n",
    "                ;;\n",
    "                \"Neff\")\n",
    "                    cmd+=\" --N-col Neff\"\n",
    "                ;;\n",
    "                \"Nca\")\n",
    "                    cmd+=\" --N-cas-col Nca\"\n",
    "                ;;\n",
    "                \"NCAS\")\n",
    "                    cmd+=\" --N-cas-col NCAS\"\n",
    "                ;;\n",
    "                \"Nco\")\n",
    "                    cmd+=\" --N-con-col Nco\"\n",
    "                ;;\n",
    "                \"Z\")\n",
    "                    cmd+=\" --signed-sumstats Z,0\"\n",
    "                    break\n",
    "                ;;\n",
    "                \"BETA\")\n",
    "                    cmd+=\" --signed-sumstats BETA,0\"\n",
    "                ;;\n",
    "                \"LogOR\")\n",
    "                    cmd+=\" --signed-sumstats LogOR,0\"\n",
    "                ;;\n",
    "                *)\n",
    "                    echo \"${arg} ignored.\"\n",
    "                ;;\n",
    "            esac\n",
    "        done\n",
    "        eval $cmd\n",
    "    fi\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "textile-format",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:big_data]",
   "language": "python",
   "name": "conda-env-big_data-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
