{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "accessory-pitch",
   "metadata": {},
   "source": [
    "# Exploring Delahaye mQTLs: Replication, Cleaning, and Enrichment\n",
    "Goals:\n",
    "1. Load in data and assess replication between Placental Regulation (Delahaye) dataset, RICHS, and Placental data from the Robinson Lab\n",
    "1. Assess effect of genotype dosage on number of significant mQTLs, repeating for sex-stratified dosage filters\n",
    "1. Assess hits that stand-up to \"fine mapping\"\n",
    "1. Assess effect sizes of sex-specific hits in terms of differences in allelic fold-change (aFC)\n",
    "1. Categorize hits based on their position relative to CpG islands and chromosome position"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "incorporate-bones",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Matplotlib created a temporary config/cache directory at /tmp/pbs.2118086.pbsha.ib.sockeye/matplotlib-fwwfofxs because the default path (/home/jovyan/.cache/matplotlib) is not a writable directory; it is highly recommended to set the MPLCONFIGDIR environment variable to a writable directory, in particular to speed up the import of Matplotlib and to better support multiprocessing.\n"
     ]
    }
   ],
   "source": [
    "import dask.dataframe as dd\n",
    "import dask.array as da\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "imperial-mirror",
   "metadata": {},
   "source": [
    "## Replication assessment\n",
    "### Loading in marginal data \n",
    "This is slightly faster in python, I have saved large datasets as a parquet file, which can be read using `pandas` and `pyarrow` in python, and the `arrow` library in R."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "hearing-seven",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 17.5 ms, sys: 12.5 ms, total: 30 ms\n",
      "Wall time: 302 ms\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "delahaye = dd.read_parquet(\"../../delahaye_QC/matrix_eqtl_data/cis_mQTL_9_methy_PC_all.parquet\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "anticipated-landscape",
   "metadata": {},
   "source": [
    "Just checking that enrichment is what I would expect for cis-mQTL analysis (mostly uniform other than many hits due to enrichment of signal in *cis*)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "impaired-bunch",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 3min, sys: 30.3 s, total: 3min 30s\n",
      "Wall time: 2min 39s\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<matplotlib.lines.Line2D at 0x2aac853fc550>"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXQAAAEDCAYAAAAlRP8qAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjMuNCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8QVMy6AAAACXBIWXMAAAsTAAALEwEAmpwYAAAS4klEQVR4nO3df4xlZZ3n8fdnu2EHVkcIXWNmu2GamQWU3QCjJSiZcdrBXbtbXWJklFZhcTEtKmZINAGNyB8Yf0TNsBN+tB3S6VWzoAtEARvJ6gQxAXYpkhZpSbO94EIN7lINo2bAjdvw3T+qcKurq+qe6j716+n3K6nknvM899zv03Xzqaefe865qSokScvfP1nsAiRJ/TDQJakRBrokNcJAl6RGGOiS1AgDXZIasaiBnmRbkmeSPNKh798k2Tnx81iSXy5AiZK0bGQxz0NP8mbgH4GvV9W/msPzPg78aVX9+3krTpKWmUWdoVfVvcBzk/cl+ZMk30/yUJIfJ3nNNE/dBNy0IEVK0jKxcrELmMZW4JKq+u9JzgKuB/7y5cYkfwScCPzdItUnSUvSkgr0JK8Azgb+c5KXd//TKd3OB26pqhcXsjZJWuqWVKAzvgT0y6o6Y5Y+5wMfW5hyJGn5WFKnLVbVr4EnkvwVQMad/nJ7klOAY4H7F6lESVqyFvu0xZsYD+dTkowmuRh4P3Bxkp8Au4BzJz1lE3BzeYtISTrAop62KEnqz5JacpEkHbxF+1B01apVtXbt2sV6eUlalh566KG9VTU0XduiBfratWsZGRlZrJeXpGUpyf+cqc0lF0lqhIEuSY0w0CWpEQMDvcstbpOsm7it7a4kP+q3RElSF11m6NuB9TM1JjmG8Rto/duq+pfAX/VSmSRpTgYG+nS3uJ3ifcBtVfXkRP9neqpNkjQHfayhnwwcm+SeiXuYXzhTxySbk4wkGRkbG+vhpSVJL+sj0FcCrwfeDrwNuDLJydN1rKqtVTVcVcNDQ9OeFy9JOkh9XFg0CuytqueB55PcC5wOPNbDsSVJHfUR6N8Frk2yEjgSOAv4mx6OO6O1V3zvd49//sW3z+dLSdKyMTDQJ25xuw5YlWQUuAo4AqCqtlTVo0m+DzwMvATcWFUznuIoSZofAwO9qjZ16PNl4Mu9VCRJOiheKSpJjTDQJakRBrokNcJAl6RGGOiS1AgDXZIaYaBLUiMMdElqhIEuSY0w0CWpEQa6JDXCQJekRhjoktQIA12SGmGgS1IjDHRJaoSBLkmNGBjoSbYleSbJrF8rl+QNSV5Mcl5/5UmSuuoyQ98OrJ+tQ5IVwJeAu3uoSZJ0EAYGelXdCzw3oNvHgVuBZ/ooSpI0d4e8hp5kNfAuYEuHvpuTjCQZGRsbO9SXliRN0seHotcAl1fVi4M6VtXWqhququGhoaEeXlqS9LKVPRxjGLg5CcAqYGOSfVX1nR6OLUnq6JADvapOfPlxku3AnYa5JC28gYGe5CZgHbAqyShwFXAEQFUNXDeXJC2MgYFeVZu6HqyqLjqkaiRJB80rRSWpEQa6JDXCQJekRhjoktQIA12SGmGgS1IjDHRJaoSBLkmNMNAlqREGuiQ1wkCXpEYY6JLUCANdkhphoEtSIwx0SWqEgS5JjTDQJakRAwM9ybYkzyR5ZIb29yd5eOLnviSn91+mJGmQLjP07cD6WdqfAP6iqk4Drga29lCXJGmOunyn6L1J1s7Sft+kzQeANT3UJUmao77X0C8G7pqpMcnmJCNJRsbGxnp+aUk6vPUW6EnewnigXz5Tn6raWlXDVTU8NDTU10tLkuiw5NJFktOAG4ENVfVsH8eUJM3NIc/Qk5wA3AZcUFWPHXpJkqSDMXCGnuQmYB2wKskocBVwBEBVbQE+CxwHXJ8EYF9VDc9XwZKk6XU5y2XTgPYPAR/qrSJJ0kHxSlFJaoSBLkmNMNAlqREGuiQ1wkCXpEYY6JLUCANdkhphoEtSIwx0SWqEgS5JjTDQJakRBrokNcJAl6RGGOiS1AgDXZIaYaBLUiMMdElqxMBAT7ItyTNJHpmhPUn+NsmeJA8neV3/ZUqSBukyQ98OrJ+lfQNw0sTPZuCGQy9LkjRXAwO9qu4Fnpuly7nA12vcA8AxSf6wrwIlSd30sYa+Gnhq0vboxD5J0gLqI9Azzb6atmOyOclIkpGxsbEeXlqS9LI+An0UOH7S9hrg6ek6VtXWqhququGhoaEeXlqS9LI+Av124MKJs13eCPyqqn7Rw3ElSXOwclCHJDcB64BVSUaBq4AjAKpqC7AD2AjsAV4APjhfxUqSZjYw0Ktq04D2Aj7WW0WSpIPilaKS1AgDXZIaYaBLUiMMdElqhIEuSY0w0CWpEQa6JDXCQJekRhjoktQIA12SGmGgS1IjDHRJaoSBLkmNMNAlqREGuiQ1wkCXpEYY6JLUCANdkhrRKdCTrE+yO8meJFdM0/6qJHck+UmSXUn8XlFJWmADAz3JCuA6YANwKrApyalTun0M+FlVnc74F0p/NcmRPdcqSZpFlxn6mcCeqnq8qn4L3AycO6VPAa9MEuAVwHPAvl4rlSTNqkugrwaemrQ9OrFvsmuB1wJPAz8F/rqqXpp6oCSbk4wkGRkbGzvIkiVJ0+kS6JlmX03ZfhuwE/jnwBnAtUl+/4AnVW2tquGqGh4aGppjqZKk2XQJ9FHg+EnbaxifiU/2QeC2GrcHeAJ4TT8lSpK66BLoDwInJTlx4oPO84Hbp/R5EjgHIMmrgVOAx/ssVJI0u5WDOlTVviSXAncDK4BtVbUrySUT7VuAq4HtSX7K+BLN5VW1dx7rliRNMTDQAapqB7Bjyr4tkx4/DfybfkuTJM2FV4pKUiMMdElqhIEuSY0w0CWpEQa6JDXCQJekRhjoktQIA12SGmGgS1IjDHRJaoSBLkmNMNAlqREGuiQ1wkCXpEYY6JLUCANdkhphoEtSIzoFepL1SXYn2ZPkihn6rEuyM8muJD/qt0xJ0iADv4IuyQrgOuBfA6PAg0lur6qfTepzDHA9sL6qnkzyB/NUryRpBl1m6GcCe6rq8ar6LXAzcO6UPu8DbquqJwGq6pl+y5QkDdIl0FcDT03aHp3YN9nJwLFJ7knyUJILpztQks1JRpKMjI2NHVzFkqRpdQn0TLOvpmyvBF4PvB14G3BlkpMPeFLV1qoarqrhoaGhORcrSZrZwDV0xmfkx0/aXgM8PU2fvVX1PPB8knuB04HHeqlSkjRQlxn6g8BJSU5MciRwPnD7lD7fBf48ycokRwNnAY/2W6okaTYDZ+hVtS/JpcDdwApgW1XtSnLJRPuWqno0yfeBh4GXgBur6pH5LFyStL8uSy5U1Q5gx5R9W6Zsfxn4cn+lSZLmwitFJakRBrokNcJAl6RGGOiS1AgDXZIa0eksl6Vs7RXf22/75198+yJVIkmLyxm6JDXCQJekRhjoktQIA12SGmGgS1IjDHRJaoSBLkmNMNAlqRHL/sKiqSZfaORFRpIOJ87QJakRBrokNaJToCdZn2R3kj1Jrpil3xuSvJjkvP5KlCR1MTDQk6wArgM2AKcCm5KcOkO/LzH+3aOSpAXW5UPRM4E9VfU4QJKbgXOBn03p93HgVuANvVZ4CPyAVNLhpMuSy2rgqUnboxP7fifJauBdwH5fHD1Vks1JRpKMjI2NzbVWSdIsugR6ptlXU7avAS6vqhdnO1BVba2q4aoaHhoa6liiJKmLLksuo8Dxk7bXAE9P6TMM3JwEYBWwMcm+qvpOH0X2weUXSa3rEugPAiclORH4e+B84H2TO1TViS8/TrIduHMphbkkHQ4GBnpV7UtyKeNnr6wAtlXVriSXTLTPum6+FDlbl9SiTpf+V9UOYMeUfdMGeVVddOhlLRzDXVIrmruXy6Ew3CUtZwb6DAx3ScvNogX67t27Wbdu3X773vOe9/DRj36UF154gY0bNx7wnIsuuoiLLrqIF1/4FWPf+cIB7a/80438s9e+mX2/HmPvnV89oP33z3wXR/+Ls/i/z47y7N3XHtD+qrPP56i1Z/Db//04z/1w6+/2/95/Gr/bwTFv/nf83prX8n9GH+WX9/5H3vjHx+33/GuuuYYzzjiDH/zgB3zuc5874Phf+9rXOOWUU7jjjjv46lcPrO8b3/gGxx9/PN/61re44YYbDmi/5ZZbWLVqFdu3b2f79u0HtO/YsYOjjz6a66+/nm9/+9sHtN9zzz0AfOUrX+HOO+/cr+2oo47irrvuAuDqq6/mhz/84X7txx13HLfeeisAn/rUp7j//vv3a1+zZg3f/OY3AbjsssvYuXPnfu0nn3wyW7eO/5tu3ryZxx57bL/2M844g2uuuQaAD3zgA4yOju7X/qY3vYkvfGH8d/7ud7+bZ599dr/2c845hyuvvBKADRs28Jvf/Ga/9ne84x188pOfBDjgfQfd33t79+7lvPMOvLPFRz7yEd773vfy1FNPccEFFxzQ/olPfIJ3vvOd7N69mw9/+MMHtH/mM5/hrW99Kzt37uSyyy47oP3zn/88Z599Nvfddx+f/vSnD2j3vbf033trr/ge/2siSyZnx6G+9yZzhn4IHnh8/1/sxv/wY4589d/zm5/v5FdT2iRpvqVq6jVCC2N4eLhGRkYO6rmTl0MORy4BSctPX8u4SR6qquHp2pyhL0Mt/EGb/Ib28wqpHwa6FsVMf5Ra+GMlLRa/4EKSGmGgS1IjDHRJaoSBLkmNMNAlqREGuiQ1wkCXpEYY6JLUCANdkhphoEtSIzoFepL1SXYn2ZPkimna35/k4Ymf+5Kc3n+pkqTZDAz0JCuA64ANwKnApiSnTun2BPAXVXUacDWwFUnSguoyQz8T2FNVj1fVb4GbgXMnd6iq+6rqHyY2HwDW9FumJGmQLoG+Gnhq0vboxL6ZXAzcdShFSZLmrsvtczPNvmm/FSPJWxgP9D+boX0zsBnghBNO6FiiJKmLLjP0UeD4SdtrgKendkpyGnAjcG5VTfv9a1W1taqGq2p4aGjoYOqVJM2gS6A/CJyU5MQkRwLnA7dP7pDkBOA24IKqemyaY0iS5tnAJZeq2pfkUuBuYAWwrap2Jblkon0L8FngOOD6JAD7ZvrOO0nS/Oj0FXRVtQPYMWXflkmPPwR8qN/SJElz4ZWiktQIA12SGmGgS1IjDHRJaoSBLkmNMNAlqREGuiQ1wkCXpEYY6JLUCANdkhphoEtSIwx0SWqEgS5JjTDQJakRBrokNcJAl6RGGOiS1AgDXZIa0SnQk6xPsjvJniRXTNOeJH870f5wktf1X6okaTYDAz3JCuA6YANwKrApyalTum0ATpr42Qzc0HOdkqQBuszQzwT2VNXjVfVb4Gbg3Cl9zgW+XuMeAI5J8oc91ypJmsXKDn1WA09N2h4FzurQZzXwi8mdkmxmfAYP8I9Jds+p2v9vFbD3IJ+7XDnmw4NjPgzkS4c05j+aqaFLoGeafXUQfaiqrcDWDq85e0HJSFUNH+pxlhPHfHhwzIeH+RpzlyWXUeD4SdtrgKcPoo8kaR51CfQHgZOSnJjkSOB84PYpfW4HLpw42+WNwK+q6hdTDyRJmj8Dl1yqal+SS4G7gRXAtqraleSSifYtwA5gI7AHeAH44PyVDPSwbLMMOebDg2M+PMzLmFN1wFK3JGkZ8kpRSWqEgS5JjVjSgX443nKgw5jfPzHWh5Pcl+T0xaizT4PGPKnfG5K8mOS8haxvPnQZc5J1SXYm2ZXkRwtdY986vLdfleSOJD+ZGPN8fxY3r5JsS/JMkkdmaO8/v6pqSf4w/gHs/wD+GDgS+Alw6pQ+G4G7GD8P/o3Af13suhdgzGcDx0483nA4jHlSv79j/AP48xa77gX4PR8D/Aw4YWL7Dxa77gUY86eBL008HgKeA45c7NoPYcxvBl4HPDJDe+/5tZRn6IfjLQcGjrmq7quqf5jYfIDxc/6Xsy6/Z4CPA7cCzyxkcfOky5jfB9xWVU8CVNVyH3eXMRfwyiQBXsF4oO9b2DL7U1X3Mj6GmfSeX0s50Ge6ncBc+ywncx3PxYz/hV/OBo45yWrgXcCWBaxrPnX5PZ8MHJvkniQPJblwwaqbH13GfC3wWsYvSvwp8NdV9dLClLcoes+vLpf+L5bebjmwjHQeT5K3MB7ofzavFc2/LmO+Bri8ql4cn7wte13GvBJ4PXAOcBRwf5IHquqx+S5unnQZ89uAncBfAn8C/JckP66qX89zbYul9/xayoF+ON5yoNN4kpwG3AhsqKpnF6i2+dJlzMPAzRNhvgrYmGRfVX1nQSrsX9f39t6qeh54Psm9wOnAcg30LmP+IPDFGl9g3pPkCeA1wH9bmBIXXO/5tZSXXA7HWw4MHHOSE4DbgAuW8WxtsoFjrqoTq2ptVa0FbgE+uozDHLq9t78L/HmSlUmOZvwOp48ucJ196jLmJxn/HwlJXg2cAjy+oFUurN7za8nO0Gtp3nJgXnUc82eB44DrJ2as+2oZ36mu45ib0mXMVfVoku8DDwMvATdW1bSnvy0HHX/PVwPbk/yU8eWIy6tq2d5WN8lNwDpgVZJR4CrgCJi//PLSf0lqxFJecpEkzYGBLkmNMNAlqREGuiQ1wkCXpEYY6JLUCANdkhrx/wA+VbFSadCekwAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%time\n",
    "hist, bins = da.histogram(delahaye['p-value'],bins=100, range=[0,1])\n",
    "hist = hist.compute()\n",
    "x = 0.5 * (bins[1:] + bins[:-1])\n",
    "width = np.diff(bins)\n",
    "plt.bar(x, hist, width);\n",
    "plt.axhline(y=hist.sum() / 100.0, linestyle='--', color='black')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aggregate-zimbabwe",
   "metadata": {
    "tags": []
   },
   "source": [
    "Loading in remaining data:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "different-egypt",
   "metadata": {},
   "outputs": [],
   "source": [
    "richs = dd.read_parquet(\"../../RICHS_QC/matrix_eqtl_data/cis_mQTL_9_methy_PC_all.parquet\")\n",
    "robinson = dd.read_parquet(\"../../robinson_results/cis_impute_mQTL_results_6_methy_PC.parquet\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "final-premises",
   "metadata": {},
   "source": [
    "### Replication of marginal signals"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "applied-gambling",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 5min 27s, sys: 57.1 s, total: 6min 24s\n",
      "Wall time: 5min 9s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "def run_replication(df1,df2):\n",
    "    return df2.merge(df1, on=['SNP','gene'])\n",
    "\n",
    "# Make data frame of hits and then filter for parallel merge\n",
    "richs_hits = richs[richs['FDR'] < 0.05].compute()\n",
    "robinson_hits =robinson[robinson['FDR'] < 0.05].compute()\n",
    "delahaye_hits = delahaye[delahaye['FDR'] < 0.05].compute()\n",
    "\n",
    "# Retrieve hits pvalues in df2 corresponding to hits in df1\n",
    "delahaye_richs = run_replication(delahaye_hits, richs)\n",
    "richs_delahaye = run_replication(richs_hits, delahaye)\n",
    "delahaye_robinson = run_replication(delahaye_hits, robinson)\n",
    "robinson_delahaye = run_replication(robinson_hits,delahaye)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "private-square",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "delahaye_richs_p = delahaye_richs['p-value_x'].compute()\n",
    "richs_delahaye_p = richs_delahaye['p-value_x'].compute()\n",
    "delahaye_robinson_p = delahaye_robinson['p-value_x'].compute()\n",
    "robinson_delahaye_p = robinson_delahaye['p-value_x'].compute()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "meaningful-appraisal",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING:rpy2.rinterface_lib.callbacks:R[write to console]: Registered S3 methods overwritten by 'ggplot2':\n",
      "  method         from \n",
      "  [.quosures     rlang\n",
      "  c.quosures     rlang\n",
      "  print.quosures rlang\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 0.7401157\n",
      "[1] 0.8158023\n",
      "[1] 0.726035\n",
      "[1] 0.8115302\n"
     ]
    }
   ],
   "source": [
    "%%R -i delahaye_richs_p,richs_delahaye_p,delahaye_robinson_p,robinson_delahaye_p\n",
    "library(qvalue)\n",
    "print(1 - pi0est(na.omit(delahaye_richs_p))$pi0)\n",
    "print(1 - pi0est(na.omit(richs_delahaye_p))$pi0)\n",
    "print(1 - pi0est(na.omit(delahaye_robinson_p))$pi0)\n",
    "print(1 - pi0est(na.omit(robinson_delahaye_p))$pi0)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "sunset-beach",
   "metadata": {},
   "source": [
    "### Loading in sex-specific signals"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "acknowledged-patrol",
   "metadata": {},
   "outputs": [],
   "source": [
    "delahaye_sex = dd.read_parquet(\"../../delahaye_QC/matrix_eqtl_data/cis_mQTL_9_methy_PC_all_sex_interaction.parquet\")\n",
    "richs_sex = dd.read_parquet(\"../../RICHS_QC/matrix_eqtl_data/cis_mQTL_9_methy_PC_all_sex_interaction.parquet\")\n",
    "robinson_sex = dd.read_parquet(\"../../robinson_results/cis_int_sex_impute_mQTL_results_6_methy_PC.parquet\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "architectural-senegal",
   "metadata": {},
   "source": [
    "The number of overlapping hits is fairly small so the $\\pi_1$ metric will not be very useful. Instead I'll just assess overlap in hits:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "tender-auditor",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "80363\n",
      "53204\n",
      "12922\n",
      "21225\n",
      "CPU times: user 13min 58s, sys: 2min 14s, total: 16min 12s\n",
      "Wall time: 9min 59s\n"
     ]
    }
   ],
   "source": [
    "%%time \n",
    "# Make data frame of hits and then filter for parallel merge\n",
    "richs_hits_sex = richs_sex[richs_sex['FDR'] < 0.05].compute()\n",
    "robinson_hits_sex =robinson_sex[robinson_sex['FDR'] < 0.05].compute()\n",
    "delahaye_hits_sex = delahaye_sex[delahaye_sex['FDR'] < 0.05].compute()\n",
    "\n",
    "delahaye_richs_sex = run_replication(delahaye_hits_sex, richs_sex).compute()\n",
    "richs_delahaye_sex = run_replication(richs_hits_sex, delahaye_sex).compute()\n",
    "delahaye_robinson_sex = run_replication(delahaye_hits_sex, robinson_sex).compute()\n",
    "robinson_delahaye_sex = run_replication(robinson_hits_sex, delahaye_sex).compute()\n",
    "\n",
    "print(len(delahaye_richs_sex))\n",
    "print(len(richs_delahaye_sex))\n",
    "print(len(delahaye_robinson_sex))\n",
    "print(len(robinson_delahaye_sex))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "exterior-season",
   "metadata": {},
   "source": [
    "The number of CpG sites with at least one sb-mQTL:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "built-assault",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20874\n",
      "11275\n",
      "3204\n",
      "3998\n"
     ]
    }
   ],
   "source": [
    "print(len(delahaye_richs_sex['gene'].unique()))\n",
    "print(len(richs_delahaye_sex['gene'].unique()))\n",
    "print(len(delahaye_robinson_sex['gene'].unique()))\n",
    "print(len(robinson_delahaye_sex['gene'].unique()))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "magnetic-history",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Effect of genotype dosage on number of mQTLs in Delahaye data\n",
    "### Marginal\n",
    "#### Loading in required data\n",
    "**Required data:**\n",
    "* Minor allele frequencies (`.afreq` file)\n",
    "* Marginal hits (computed on dask dataframe)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "binding-relationship",
   "metadata": {},
   "outputs": [],
   "source": [
    "delahaye_maf = pd.read_csv(\"../../delahaye_QC/placenta_regulatory_landscape/RootStudyConsentSet_phs001717.PlacentalRegulation.v1.p1.c1.HMB-IRB-PUB-COL-MDS/genotype_qc/all_imputed_r2_30_rsid.afreq\",sep='\\s+')\n",
    "if not \"delahaye_hits\" in vars():\n",
    "    delahaye_hits = delahaye[delahaye['FDR'] < 0.05].compute()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "adverse-regard",
   "metadata": {},
   "source": [
    "#### Filtering\n",
    "By default `plink` filters minor allele frequency (MAF) at one end. I.e. if I filter using the command\n",
    "```bash\n",
    "plink --bfile data\\\n",
    "    --maf 0.01\\\n",
    "    --make-bed\\\n",
    "    --out data_filtered\n",
    "```\n",
    "the resulting files will still contain SNPs with MAF > 0.99. I didn't account for this when running `matrixEQTL` to run associations so I'm going to account for it now, and test MAF thresholds at 0.05 as well."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "catholic-telescope",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Delahaye hits: 8712448\n",
      "Delahaye hits at true MAF > 0.01 : 7649972\n",
      "Delahaye hits at true MAF > 0.05 : 6498652\n",
      "CpGs with one hit in Delahaye: 275806\n",
      "CpGs with one hit in Delahaye at true MAF > 0.01 : 203617\n",
      "CpGs with one hit in Delahaye at true MAF > 0.05 : 164780\n"
     ]
    }
   ],
   "source": [
    "# Get SNPs at specified threshold\n",
    "maf01 = delahaye_maf[(delahaye_maf['ALT_FREQS'] >= 0.01) & (delahaye_maf['ALT_FREQS'] <= 0.99)]['ID']\n",
    "maf05 = delahaye_maf[(delahaye_maf['ALT_FREQS'] >= 0.05) & (delahaye_maf['ALT_FREQS'] <= 0.95)]['ID']\n",
    "\n",
    "# Filter hits\n",
    "print(f\"Delahaye hits: {len(delahaye_hits)}\")\n",
    "print(f\"Delahaye hits at true MAF > 0.01 : {len(delahaye_hits[delahaye_hits['SNP'].isin(maf01)])}\")\n",
    "print(f\"Delahaye hits at true MAF > 0.05 : {len(delahaye_hits[delahaye_hits['SNP'].isin(maf05)])}\")\n",
    "# Unique\n",
    "print(f\"CpGs with one hit in Delahaye: {len(delahaye_hits['gene'].unique())}\")\n",
    "print(f\"CpGs with one hit in Delahaye at true MAF > 0.01 : {len(delahaye_hits[delahaye_hits['SNP'].isin(maf01)]['gene'].unique())}\")\n",
    "print(f\"CpGs with one hit in Delahaye at true MAF > 0.05 : {len(delahaye_hits[delahaye_hits['SNP'].isin(maf05)]['gene'].unique())}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "contemporary-recycling",
   "metadata": {},
   "source": [
    "### Sex-specific\n",
    "#### Loading in required data\n",
    "**Required data:**\n",
    "* Genotypes for sex-specific hits (for sex specific minor allele frequencies)\n",
    "* Sex labels\n",
    "* Sex specific hits (computed on dask dataframe)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "downtown-practitioner",
   "metadata": {},
   "outputs": [],
   "source": [
    "if not \"delahaye_hits_sex\" in vars():\n",
    "    delahaye_hits_sex = delahaye_sex[delahaye_sex['FDR'] < 0.05].compute()\n",
    "\n",
    "# Filter SNPs to those in hits\n",
    "genotypes = dd.read_csv(\"../../delahaye_QC/matrix_eqtl_data/all_imputed_matrixeQTL.txt\", sep='\\s+')\n",
    "genotypes = genotypes[genotypes['SNP'].isin(delahaye_hits_sex['SNP'])].compute()\n",
    "\n",
    "# Read in metadata\n",
    "metadata = pd.read_csv(\"../../delahaye_QC/matrix_eqtl_data/mQTL_covar_9_methy_PC.txt\", sep='\\s+')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "black-germany",
   "metadata": {},
   "source": [
    "#### Filtering\n",
    "The filters to try here are:\n",
    "1. Global MAF filters of 0.01, 0.05 (tested on marginal hits in previous section)\n",
    "1. Dosage filters requireing 0.01, 0.05 subjects in each dosage group\n",
    "1. Sex specific MAF and dosage filters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "id": "willing-shepherd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>group</th>\n",
       "      <th>MAF</th>\n",
       "      <th>True MAF</th>\n",
       "      <th>dosage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>all</td>\n",
       "      <td>0.01</td>\n",
       "      <td>4616</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>males</td>\n",
       "      <td>0.01</td>\n",
       "      <td>4595</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>females</td>\n",
       "      <td>0.01</td>\n",
       "      <td>4576</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>all</td>\n",
       "      <td>0.05</td>\n",
       "      <td>3984</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>males</td>\n",
       "      <td>0.05</td>\n",
       "      <td>3797</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>females</td>\n",
       "      <td>0.05</td>\n",
       "      <td>3886</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     group   MAF  True MAF  dosage\n",
       "0      all  0.01      4616       0\n",
       "1    males  0.01      4595       0\n",
       "2  females  0.01      4576       0\n",
       "3      all  0.05      3984       0\n",
       "4    males  0.05      3797       0\n",
       "5  females  0.05      3886       0"
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def maf_filter(geno, N, maf): # SNP, genotypes, subjects\n",
    "    freq = ((2 * geno.iloc[:,1:].sum(axis=1)) / N).to_numpy()\n",
    "    return geno['SNP'][(freq >= maf) & (freq <= (1.0 - maf))]\n",
    "\n",
    "def dosage_filter(dosage_freq, N, maf):\n",
    "    filt = dosage_freq / N\n",
    "    filt0 = (filt[[0]] <= maf).to_numpy().flatten() & (filt[[0]] >= (1.0 - maf)).to_numpy().flatten()\n",
    "    filt1 = (filt[[1]] <= maf).to_numpy().flatten() & (filt[[1]] >= (1.0 - maf)).to_numpy().flatten()\n",
    "    filt2 = (filt[[2]] <= maf).to_numpy().flatten() & (filt[[2]] >= (1.0 - maf)).to_numpy().flatten()\n",
    "    return dosage_freq.index[filt0 & filt1 & filt2]\n",
    "\n",
    "male_ix = metadata[metadata['id'] == \"Sex\"].iloc[:,1:] == 1\n",
    "female_ix = metadata[metadata['id'] == \"Sex\"].iloc[:,1:] == 0\n",
    "males = ['SNP'] + metadata.columns[1:][male_ix.to_numpy().flatten()].tolist()\n",
    "females = ['SNP'] + metadata.columns[1:][female_ix.to_numpy().flatten()].tolist()\n",
    "\n",
    "\n",
    "all_01 = maf_filter(genotypes, len(genotypes.columns) - 1, 0.01)\n",
    "all_05 = maf_filter(genotypes, len(genotypes.columns) - 1, 0.05)\n",
    "male_01 = maf_filter(genotypes[males], len(males) - 1, 0.01)\n",
    "male_05 = maf_filter(genotypes[males], len(males) - 1, 0.05)\n",
    "female_01 = maf_filter(genotypes[females], len(females) - 1, 0.01)\n",
    "female_05 = maf_filter(genotypes[females], len(females) - 1, 0.05)\n",
    "\n",
    "\n",
    "# Dosage specific filtering\n",
    "enum = genotypes.round().melt('SNP')\n",
    "male_enum = genotypes[males].round().melt('SNP')\n",
    "female_enum = genotypes[females].round().melt('SNP')\n",
    "\n",
    "dosage = pd.crosstab(enum['SNP'], enum['value'])\n",
    "male_dosage = pd.crosstab(male_enum['SNP'], male_enum['value'])\n",
    "female_dosage = pd.crosstab(female_enum['SNP'], female_enum['value'])\n",
    "\n",
    "all_dosage_01 = dosage_filter(dosage, len(genotypes.columns) - 1, 0.01)\n",
    "all_dosage_05 = dosage_filter(dosage, len(genotypes.columns) - 1, 0.05)\n",
    "male_dosage_01 = dosage_filter(male_dosage, len(males) - 1, 0.01)\n",
    "male_dosage_05 = dosage_filter(male_dosage, len(males) - 1, 0.05)\n",
    "female_dosage_01 = dosage_filter(female_dosage, len(females) - 1, 0.01)\n",
    "female_dosage_05 = dosage_filter(female_dosage, len(females) - 1, 0.05)\n",
    "\n",
    "# Print a nice table\n",
    "result = pd.DataFrame(\n",
    "    { \n",
    "        'group': ['all', 'males', 'females','all', 'males', 'females'],\n",
    "        'MAF' : [0.01,0.01,0.01,0.05,0.05,0.05],\n",
    "        'True MAF': [len(all_01), len(male_01), len(female_01),len(all_05),len(male_05),len(female_05)],\n",
    "        'dosage': [len(all_dosage_01), len(male_dosage_01), len(female_dosage_01),len(all_dosage_05),len(male_dosage_05),len(female_dosage_05)]\n",
    "    }\n",
    ")\n",
    "result"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "committed-adrian",
   "metadata": {},
   "source": [
    "## Fine mapping mQTLs with CAVIAR"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:big_data]",
   "language": "python",
   "name": "conda-env-big_data-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  },
  "toc-autonumbering": false,
  "toc-showcode": false,
  "toc-showmarkdowntxt": false,
  "toc-showtags": false
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
