{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "aaefee8b-a24d-4c47-8832-aa497646ad4d",
   "metadata": {
    "tags": []
   },
   "source": [
    "# Colocalization between placental mQTL, psychiatric and developmental traits\n",
    "\n",
    "Idea: for all heritable CpGs, test for colocalization between traits and GWAS summary statistics for conditions where a significant proportion of heritability can be explained by placental mQTL.\n",
    "Traits to include:\n",
    "* Neuropsychiatric\n",
    "    1. OCD\n",
    "    1. TS\n",
    "    1. ADHD\n",
    "    1. ASD\n",
    "    1. MDD\n",
    "    1. BIP\n",
    "    1. AN\n",
    "    1. SCZ\n",
    "* Developmental: \n",
    "    1. Pubertal Growth Start\n",
    "    1. Asthma\n",
    "    1. Child BMI\n",
    "    1. Child Onset Asthma"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "91e6db99-806a-49f2-a72f-63d3b0b20b93",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "── \u001b[1mAttaching packages\u001b[22m ─────────────────────────────────────── tidyverse 1.3.0 ──\n",
      "\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2\u001b[39m 3.3.3     \u001b[32m✔\u001b[39m \u001b[34mpurrr  \u001b[39m 0.3.4\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtibble \u001b[39m 3.1.0     \u001b[32m✔\u001b[39m \u001b[34mdplyr  \u001b[39m 1.0.4\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtidyr  \u001b[39m 1.1.3     \u001b[32m✔\u001b[39m \u001b[34mstringr\u001b[39m 1.4.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mreadr  \u001b[39m 1.4.0     \u001b[32m✔\u001b[39m \u001b[34mforcats\u001b[39m 0.5.1\n",
      "\n",
      "── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m    masks \u001b[34mstats\u001b[39m::lag()\n",
      "\n",
      "\n",
      "Attaching package: ‘data.table’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:purrr’:\n",
      "\n",
      "    transpose\n",
      "\n",
      "\n",
      "This is a new update to coloc.\n",
      "\n",
      "Loading required package: bigstatsr\n",
      "\n",
      "here() starts at /scratch/st-dennisjk-1/wcasazza/sex_specific_mQTL\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(tidyverse)\n",
    "library(data.table)\n",
    "library(coloc)\n",
    "library(bigsnpr)\n",
    "library(here)\n",
    "library(parallel)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "9a458f17-686e-4176-ad0f-70790f5cef81",
   "metadata": {},
   "outputs": [],
   "source": [
    "if (file.exists(here(\"data\",\"marginal_mcpg_bonf.txt.gz\"))){\n",
    "    marginal_bonf <- fread(here(\"data\",\"marginal_mcpg_bonf.txt.gz\"),key=\"SNP\")\n",
    "}else{\n",
    "    marginal <- fread(here(\"data\",\"delahaye_richs_marginal_mecs_maf01.txt.gz\"))\n",
    "    marginal_bonf <- fread(here(\"data\",\"delahaye_richs_marginal_bonf_maf05.txt.gz\"))[Chr != 23]\n",
    "    marginal_bonf <- marginal[Probe %in%marginal_bonf$Probe]\n",
    "    fwrite(marginal_bonf,here(\"data\",\"marginal_mcpg_bonf.txt.gz\"),row.names=F,quote=F,sep='\\t')\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "5c2a046c-964e-4c67-b714-2ce1f6ee512b",
   "metadata": {},
   "outputs": [],
   "source": [
    "rds < -snp_readBed2(\n",
    "    \"/arc/project/st-dennisjk-1/shared/data/1000G_EUR_ldsc_data/1000G_EUR_Phase3_plink/1000G.EUR.QC.ALL.bed\",\n",
    "    backingfile=tempfile(),\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6272cacd-9824-4a90-97d8-475dfc8e6865",
   "metadata": {},
   "outputs": [],
   "source": [
    "reference < -snp_attach(rds)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "b95cfedd-0486-46ff-b962-966ec72c0cc6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "List of 3\n",
      " $ genotypes:Reference class 'FBM.code256' [package \"bigstatsr\"] with 16 fields\n",
      "  ..$ extptr      :<externalptr> \n",
      "  ..$ extptr_rw   :<externalptr> \n",
      "  ..$ nrow        : int 489\n",
      "  ..$ ncol        : int 9997231\n",
      "  ..$ type        : Named int 1\n",
      "  .. ..- attr(*, \"names\")= chr \"unsigned char\"\n",
      "  ..$ backingfile : chr \"/tmp/pbs.3124043.pbsha.ib.sockeye/Rtmp2tC1yx/file11da523a0fad.bk\"\n",
      "  ..$ is_read_only: logi FALSE\n",
      "  ..$ address     :<externalptr> \n",
      "  ..$ address_rw  :<externalptr> \n",
      "  ..$ bk          : chr \"/tmp/pbs.3124043.pbsha.ib.sockeye/Rtmp2tC1yx/file11da523a0fad.bk\"\n",
      "  ..$ rds         : chr \"/tmp/pbs.3124043.pbsha.ib.sockeye/Rtmp2tC1yx/file11da523a0fad.rds\"\n",
      "  ..$ is_saved    : logi TRUE\n",
      "  ..$ type_chr    : chr \"unsigned char\"\n",
      "  ..$ type_size   : int 1\n",
      "  ..$ file_size   : num 4.89e+09\n",
      "  ..$ code256     : num [1:256] 0 1 2 NA NA NA NA NA NA NA ...\n",
      "  ..and 26 methods, of which 12 are  possibly relevant:\n",
      "  ..  add_columns, as.FBM, bm, bm.desc, check_dimensions,\n",
      "  ..  check_write_permissions, copy#envRefClass, initialize, initialize#FBM,\n",
      "  ..  save, show#envRefClass, show#FBM\n",
      " $ fam      :'data.frame':\t489 obs. of  6 variables:\n",
      "  ..$ family.ID  : chr [1:489] \"HG00096\" \"HG00097\" \"HG00099\" \"HG00101\" ...\n",
      "  ..$ sample.ID  : chr [1:489] \"HG00096\" \"HG00097\" \"HG00099\" \"HG00101\" ...\n",
      "  ..$ paternal.ID: int [1:489] 0 0 0 0 0 0 0 0 0 0 ...\n",
      "  ..$ maternal.ID: int [1:489] 0 0 0 0 0 0 0 0 0 0 ...\n",
      "  ..$ sex        : int [1:489] 0 0 0 0 0 0 0 0 0 0 ...\n",
      "  ..$ affection  : int [1:489] -9 -9 -9 -9 -9 -9 -9 -9 -9 -9 ...\n",
      " $ map      :'data.frame':\t9997231 obs. of  6 variables:\n",
      "  ..$ chromosome  : int [1:9997231] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  ..$ marker.ID   : chr [1:9997231] \"rs575272151\" \"rs544419019\" \"rs540538026\" \"rs62635286\" ...\n",
      "  ..$ genetic.dist: num [1:9997231] 0 0 0 0 0 0 0 0 0 0 ...\n",
      "  ..$ physical.pos: int [1:9997231] 11008 11012 13110 13116 13118 13273 13550 14464 14599 14604 ...\n",
      "  ..$ allele1     : chr [1:9997231] \"G\" \"G\" \"A\" \"G\" ...\n",
      "  ..$ allele2     : chr [1:9997231] \"C\" \"C\" \"G\" \"T\" ...\n",
      " - attr(*, \"class\")= chr \"bigSNP\"\n"
     ]
    }
   ],
   "source": [
    "str(reference)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "8c1a0c0a-ffc5-413d-81e5-9c6bcdd8e7d1",
   "metadata": {},
   "outputs": [],
   "source": [
    "compute_coloc <- function(SNP,mqtl,gwas,method=\"coloc\", type=\"quant\",s=NULL){# SNP must be in SNP column of mqtl and gwas\n",
    "    D1 <- list(\n",
    "            beta = mqtl$b,\n",
    "            varbeta = mqtl$SE^2,\n",
    "            snp = mqtl$SNP,\n",
    "            position = mqtl$BP,\n",
    "            N = 400,\n",
    "            MAF = mqtl$Freq,\n",
    "            type=\"quant\"\n",
    "        )\n",
    "    D2 <- list(\n",
    "            pvalues = pnorm(-abs(gwas$Z))*2,\n",
    "            z=gwas$Z,\n",
    "            snp = gwas$SNP,\n",
    "            MAF = snp_MAF(reference$genotypes,ind.col = match(gwas$SNP,reference$map$marker.ID)),\n",
    "            N = min(gwas$N),\n",
    "            type=type,\n",
    "            s = s\n",
    "        )\n",
    "    if(method == \"susie\"){\n",
    "        LD <- snp_cor(reference$genotypes,ind.col = match(SNP, reference$map$marker.ID))^2\n",
    "        LD <- as.matrix(LD)\n",
    "        colnames(LD) <- SNP\n",
    "        rownames(LD) <- SNP\n",
    "        D1$LD <- LD\n",
    "        D2$LD <- LD\n",
    "        S1 <- runsusie(D1)\n",
    "        S2 <- runsusie(D2)\n",
    "        return(coloc.susie(S1,S2))\n",
    "    }else if(method == \"coloc\"){\n",
    "        return(coloc.abf(D1,D2))\n",
    "    }else{\n",
    "        return(NULL)\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "4cb399ba-acbe-4cee-9661-f73cfee7108e",
   "metadata": {},
   "outputs": [],
   "source": [
    "gwas <- fread(\"../../tmp_GWAS/pgc_formatted_sumstats/adhd_jul2017.sumstats.gz\",key=\"SNP\")\n",
    "\n",
    "tmp_marginal_bonf <- marginal_bonf[intersect(gwas$SNP,reference$map$marker.ID),on=\"SNP\",nomatch=0]\n",
    "\n",
    "eligible_cpg <- unlist(mclapply(\n",
    "    unique(tmp_marginal_bonf$Probe),\n",
    "    function(probe){\n",
    "        mqtl <- tmp_marginal_bonf[Probe == probe]\n",
    "        gwas_tmp <- gwas[mqtl$SNP,on=\"SNP\"]\n",
    "        return(min(mqtl$p) < 5e-8 & max(abs(gwas_tmp$Z)) > 5.45 )\n",
    "    },\n",
    "    mc.cores=16\n",
    "))\n",
    "test <- mclapply(\n",
    "    unique(marginal_bonf$Probe)[eligible_cpg],\n",
    "    function(probe){\n",
    "        mqtl <- tmp_marginal_bonf[Probe == probe]\n",
    "        gwas_tmp <- gwas[SNP %in% mqtl$SNP]\n",
    "        res <- suppressMessages(compute_coloc(mqtl$SNP,mqtl,gwas_tmp,method=\"coloc\",s=0.36,type=\"cc\")$summary)\n",
    "        return(res)\n",
    "    },\n",
    "    mc.cores=16\n",
    ")\n",
    "\n",
    "names(test) <- unique(marginal_bonf$Probe)[eligible_cpg]\n",
    "result_adhd <- rbindlist(lapply(test,function(x)data.table(t(x))))\n",
    "result_adhd"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6d28b2a3-a700-441c-a54f-d171ab39ab87",
   "metadata": {},
   "source": [
    "# Repeating above for all psychiatric GWAS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "fb406121-708f-47e4-ba50-3bf09ee68d18",
   "metadata": {},
   "outputs": [],
   "source": [
    "sumstat_files < -c(\n",
    "    \"/scratch/st-dennisjk-1/wcasazza/tmp_GWAS/pgc_formatted_sumstats/adhd_jul2017.sumstats.gz\",\n",
    "    \"/scratch/st-dennisjk-1/wcasazza/tmp_GWAS/pgc_formatted_sumstats/anxiety.meta.full.fs.tbl.sumstats.gz\",\n",
    "    \"/scratch/st-dennisjk-1/wcasazza/tmp_GWAS/pgc_formatted_sumstats/anxiety.meta.full.cc.tbl.sumstats.gz\",\n",
    "    \"/scratch/st-dennisjk-1/wcasazza/tmp_GWAS/pgc_formatted_sumstats/pgcAN2.2019-07_refmt.vcf.tsv.sumstats.gz\",\n",
    "    \"/scratch/st-dennisjk-1/wcasazza/tmp_GWAS/pgc_formatted_sumstats/AUDIT_UKB_2018_AJP_fixed.txt.sumstats.gz\",\n",
    "    \"/scratch/st-dennisjk-1/wcasazza/tmp_GWAS/pgc_formatted_sumstats/Cannabis_ICC_23andmetop_UKB_het_fixed.txt.sumstats.gz\",\n",
    "    \"/scratch/st-dennisjk-1/wcasazza/tmp_GWAS/pgc_formatted_sumstats/iPSYCH-PGC_ASD_Nov2017.sumstats.gz\",\n",
    "    \"/scratch/st-dennisjk-1/wcasazza/tmp_GWAS/pgc_formatted_sumstats/ocd_aug2017.sumstats.gz\",\n",
    "    \"/scratch/st-dennisjk-1/wcasazza/tmp_GWAS/pgc_formatted_sumstats/pgc_alcdep.eur_unrelated.aug2018_release_refmt.txt.sumstats.gz\",\n",
    "    \"/scratch/st-dennisjk-1/wcasazza/tmp_GWAS/pgc_formatted_sumstats/pgc_bip_2018.sumstats.gz\",\n",
    "    \"/scratch/st-dennisjk-1/wcasazza/tmp_GWAS/pgc_formatted_sumstats/PGC_UKB_depression_genome-wide_fixed.txt.sumstats.gz\",\n",
    "    \"/scratch/st-dennisjk-1/wcasazza/tmp_GWAS/pgc_formatted_sumstats/MDD2018_ex23andMe.sumstats.gz\",\n",
    "    \"/scratch/st-dennisjk-1/wcasazza/tmp_GWAS/pgc_formatted_sumstats/PGC3_SCZ_wave3_public.v2.tsv.sumstats.gz\",\n",
    "    \"/scratch/st-dennisjk-1/wcasazza/tmp_GWAS/pgc_formatted_sumstats/tag.cpd.tbl.sumstats.gz\",\n",
    "    \"/scratch/st-dennisjk-1/wcasazza/tmp_GWAS/pgc_formatted_sumstats/TS_Oct2018.sumstats.gz\",\n",
    "    \"/scratch/st-dennisjk-1/wcasazza/tmp_GWAS/pgc_formatted_sumstats/pts_all_freeze2_overall.results.sumstats.gz\",\n",
    ")\n",
    "trait_names < -c(\n",
    "    \"ADHD\",\n",
    "    \"ANXFS\",\n",
    "    \"ANX\",\n",
    "    \"AN\",\n",
    "    \"AUDIT\",\n",
    "    \"CUD\",\n",
    "    \"ASD\",\n",
    "    \"OCD\",\n",
    "    \"ALC\",\n",
    "    \"BIP\",\n",
    "    \"MDD\",\n",
    "    \"MDD2018\",\n",
    "    \"SCZ\",\n",
    "    \"TAG_CPD\",\n",
    "    \"TS\",\n",
    "    \"PTSD\",\n",
    ")\n",
    "sample_prev < -c(\n",
    "    0.36,\n",
    "    NULL,\n",
    "    0.33,\n",
    "    0.23,\n",
    "    NULL,\n",
    "    0.30,\n",
    "    0.40,\n",
    "    0.28,\n",
    "    0.26,\n",
    "    0.15,\n",
    "    0.34,\n",
    "    0.35,\n",
    "    0.42,\n",
    "    NULL,\n",
    "    0.34,\n",
    "    0.15,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4b24326c-d696-44c0-87b4-88a27ee5d54b",
   "metadata": {},
   "outputs": [],
   "source": [
    "result <- list()\n",
    "for( i in 1:length(sumstat_files)){\n",
    "    gwas <- fread(sumstat_files[i])\n",
    "    tmp_marginal_bonf <- marginal_bonf[intersect(gwas$SNP,reference$map$marker.ID),on=\"SNP\",nomatch=0]\n",
    "    eligible_cpg <- unlist(mclapply(\n",
    "        unique(tmp_marginal_bonf$Probe),\n",
    "        function(probe){\n",
    "            mqtl <- tmp_marginal_bonf[Probe == probe]\n",
    "            gwas_tmp <- gwas[mqtl$SNP,on=\"SNP\"]\n",
    "            return(min(mqtl$p) < 5e-8 & max(abs(gwas_tmp$Z)) > 5.45 )\n",
    "        },\n",
    "        mc.cores=16\n",
    "    ))\n",
    "    if(sum(eligible_cpg) ){\n",
    "        test <- mclapply(\n",
    "            unique(tmp_marginal_bonf$Probe)[eligible_cpg],\n",
    "            function(probe){\n",
    "                mqtl <- tmp_marginal_bonf[Probe == probe]\n",
    "                gwas_tmp <- gwas[SNP %in% mqtl$SNP]\n",
    "                res <- suppressMessages(\n",
    "                    compute_coloc(\n",
    "                        mqtl$SNP,\n",
    "                        mqtl,\n",
    "                        gwas_tmp,\n",
    "                        method=\"coloc\",\n",
    "                        s=sample_prev[i],\n",
    "                        type=ifelse(is.null(sample_prev[i]),\"quant\",\"cc\")\n",
    "                    )$summary\n",
    "                )\n",
    "                return(res)\n",
    "            },\n",
    "            mc.cores=16\n",
    "        )\n",
    "        names(test) <- unique(tmp_marginal_bonf$Probe)[eligible_cpg]\n",
    "        res <- rbindlist(lapply(test,function(x)data.table(t(x))))\n",
    "        result[[trait_names[i]]] <- res\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "054a0c93-4e35-492f-b600-acc38cf715a8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " list()\n"
     ]
    }
   ],
   "source": [
    "str(result)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "388f74ef-1165-4e73-aad4-b3e387c71ab6",
   "metadata": {},
   "source": [
    "# Neonatal Traits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5b05266d-c0c6-4e48-bb6e-f609d53e2abb",
   "metadata": {},
   "outputs": [],
   "source": [
    "sumstat_files < -c(\n",
    "    \"/scratch/st-dennisjk-1/wcasazza/tmp_GWAS/neonatal_gwas/formatted/Pubertal_growth_PGF_PGM_combined.txt.sumstats.gz\",\n",
    "    \"/scratch/st-dennisjk-1/wcasazza/tmp_GWAS/neonatal_gwas/formatted/EGG_Obesity_Meta_Analysis_1.txt.sumstats.gz\",\n",
    "    \"/scratch/st-dennisjk-1/wcasazza/tmp_GWAS/neonatal_gwas/formatted/age_hay_fever_irnt.gwas.imputed_v3.both_sexes.tsv.sumstats.gz\",\n",
    "    \"/scratch/st-dennisjk-1/wcasazza/tmp_GWAS/neonatal_gwas/formatted/EGG-TotalGWG-Offspring.txt.sumstats.gz\",\n",
    "    \"/scratch/st-dennisjk-1/wcasazza/tmp_GWAS/neonatal_gwas/formatted/Pubertal_growth_PTF_PTM_combined.txt.sumstats.gz\",\n",
    "    \"/scratch/st-dennisjk-1/wcasazza/tmp_GWAS/neonatal_gwas/formatted/age_hay_fever_raw.gwas.imputed_v3.both_sexes.tsv.sumstats.gz\",\n",
    "    \"/scratch/st-dennisjk-1/wcasazza/tmp_GWAS/neonatal_gwas/formatted/age_asthma_raw.gwas.imputed_v3.both_sexes.tsv.sumstats.gz\",\n",
    "    \"/scratch/st-dennisjk-1/wcasazza/tmp_GWAS/neonatal_gwas/formatted/age_asthma_irnt.gwas.imputed_v3.both_sexes.tsv.sumstats.gz\",\n",
    "    \"/scratch/st-dennisjk-1/wcasazza/tmp_GWAS/neonatal_gwas/formatted/age_diabetes_raw.gwas.imputed_v3.both_sexes.tsv.sumstats.gz\",\n",
    "    \"/scratch/st-dennisjk-1/wcasazza/tmp_GWAS/neonatal_gwas/formatted/age_diabetes_irnt.gwas.imputed_v3.both_sexes.tsv.sumstats.gz\",\n",
    "    \"/scratch/st-dennisjk-1/wcasazza/tmp_GWAS/neonatal_gwas/formatted/EGG_BMI_HapMap_DISCOVERY.txt.sumstats.gz\",\n",
    "    \"/scratch/st-dennisjk-1/wcasazza/tmp_GWAS/neonatal_gwas/formatted/Pubertal_growth_10F_12M_combined.txt.sumstats.gz\",\n",
    "    \"/scratch/st-dennisjk-1/wcasazza/tmp_GWAS/neonatal_gwas/formatted/EGG_HC_DISCOVERY.v2.txt.sumstats.gz\",\n",
    "    \"/scratch/st-dennisjk-1/wcasazza/tmp_GWAS/neonatal_gwas/formatted/CHILD_ONSET_ASTHMA.20180501.allchr.assoc.GC.sumstats.gz\",\n",
    ")\n",
    "trait_names < -c(\n",
    "    \"ADHD\",\n",
    "    \"ANXFS\",\n",
    "    \"ANX\",\n",
    "    \"AN\",\n",
    "    \"AUDIT\",\n",
    "    \"CUD\",\n",
    "    \"ASD\",\n",
    "    \"OCD\",\n",
    "    \"ALC\",\n",
    "    \"BIP\",\n",
    "    \"MDD\",\n",
    "    \"MDD2018\",\n",
    "    \"SCZ\",\n",
    "    \"TAG_CPD\",\n",
    "    \"TS\",\n",
    "    \"PTSD\",\n",
    ")\n",
    "sample_prev < -c(\n",
    "    NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, 0.0296\n",
    ")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.0.3"
  },
  "widgets": {
   "application/vnd.jupyter.widget-state+json": {
    "state": {},
    "version_major": 2,
    "version_minor": 0
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
