---
title: "Delahaye Post mQTL analyses"
output: github_document
---

```{r setup, include=FALSE}
library(tidyverse)
library(data.table)
library(qvalue)
setwd("/scratch/st-dennisjk-1/wcasazza/delahaye_QC/")
knitr::opts_knit$set(root.dir = "/scratch/st-dennisjk-1/wcasazza/delahaye_QC/")
knitr::opts_chunk$set(echo = TRUE,cache.lazy = FALSE,cache = FALSE,cache.path = "global_cache")
```
```{r marginal hits}
system.time(opt_pc <- fread("matrix_eqtl_data/cis_mQTL_9_methy_PC_all.txt"))
delahaye <- fread("delahaye_2020_mQTL.csv")
hist(opt_pc$`p-value`, breaks = 100)
sig_delahaye <- opt_pc[FDR < 0.05]
```
## Replication
Delahaye and itself:
```{r mqtl replication}
delahaye_repeat_merged <-merge(opt_pc,delahaye, by.x = c("SNP", "gene"), by.y = c("SNPID", "cpgID"))
tryCatch(1 - pi0est(delahaye_repeat_merged$`p-value`)$pi0,"inf")
```

Robinson and Delahaye
```{r robinson loading}
robinson_mqtl <- fread("../robinson_results/cis_impute_mQTL_results_6_methy_PC.txt")
```

```{r robinson replication}
robinson_matched <- robinson_mqtl %>% left_join(sig_delahaye, by = c("SNP", "gene"))
1 - pi0est(robinson_matched$`p-value.x`, m = nrow(robinson_matched))$pi0
```
delahaye and robinson:
```{r delahaye robinson}
robinson_sig <- robinson_mqtl[FDR < 0.05]
delahaye_matched <- merge(opt_pc, robinson_sig, by = c("SNP", "gene"))
1 - pi0est(delahaye_matched$`p-value.x`, m = nrow(delahaye_matched))$pi0
```

Delahaye and RICHS
```{r richs mqtl load}
richs_mqtl <- fread("../RICHS_QC/matrix_eqtl_data/cis_mQTL_9_methy_PC_all.txt")
```
```{r richs delahaye pi1}

richs_matched <- merge(richs_mqtl, sig_delahaye, by = c("SNP", "gene"))
1 - pi0est(richs_matched$`p-value.x`, m = nrow(richs_matched))$pi0
```



## Sex specific mQTL
```{r sex specific mqtl load}
sex_spec <- fread("matrix_eqtl_data/cis_mQTL_9_methy_PC_all_sex_interaction.txt")
sig_sex <- sex_spec[FDR < 0.05]
```
```{r robinson sex specific hits}
robinson_sex <- rbind(
  fread("../robinson_results/cis_int_sex_impute_mQTL_results_6_methy_PC.txt"),
  fread("../robinson_results/cis_pre_impute_mQTL_results_6_methy_PC_sex_interaction_chrX.txt")
)
```

```{r}
robinson_matched <- merge(robinson_sex, sig_sex, by = c("SNP", "gene"))
1 - pi0est(robinson_matched$`p-value.x`, m = nrow(robinson_matched))$pi0
robinson_matched[FDR.x < 0.05]
ggplot(robinson_matched, aes(-log10(`p-value.x`), -log10(`p-value.y`))) +
  geom_point() +
  labs(x = "Robinson Interaction", y = "Delahaye Interaction")
```
delahaye and robinson:
```{r}
robinson_sig <- robinson_sex[FDR < 0.05]
delahaye_matched <- merge(sex_spec, robinson_sig, by = c("SNP", "gene"))
1 - qvalue::pi0est(delahaye_matched$`p-value.x`, m = nrow(delahaye_matched))$pi0
delahaye_matched[FDR.x < 0.05]
ggplot(delahaye_matched, aes(-log10(`p-value.x`), -log10(`p-value.y`))) +
  geom_point() +
  labs(x = "Delahaye Interaction", y = "Robinson Interaction") +
  coord_flip()
```

Delahaye and RICHS
```{r}
richs_sex <- fread("../RICHS_QC/matrix_eqtl_data/cis_mQTL_9_methy_PC_all_sex_interaction.txt")
```
```{r}
richs_matched <- merge(richs_sex, sig_sex, by = c("SNP", "gene"))
1 - qvalue::pi0est(richs_matched$`p-value.x`)$pi0
richs_matched[FDR.x < 0.05]
fwrite(richs_matched[FDR.x < 0.05], "richs_delahaye_matched_sbmqtl_fdr_05.txt", quote = F)
```
```{r}
richs_sig <- richs_sex[FDR < 0.05]
delahaye_matched <- merge(sex_spec, richs_sig, by = c("SNP", "gene"))
1 - qvalue::pi0est(delahaye_matched$`p-value.x`, m = nrow(delahaye_matched))$pi0
delahaye_matched[FDR.x < 0.05]
ggplot(delahaye_matched, aes(-log10(`p-value.x`), -log10(`p-value.y`))) +
  geom_point() +
  labs(x = "Delahaye Interaction", y = "RICHS Interaction") +
  coord_flip()
```
## Thresholding Delahaye QTLs
<!-- @todo implement QTL Quality filters -->
<!-- @body delahaye et al 2018 use FDR based on 1000 permutations to narrow down their initial number, they then restrict 5% of each allelic dosage. I should report something at least including the dosage (especially in the sex specific hits) and perhaps a more stringent cutoff like bonferroni or a lower FDR. Another thing I should do is report those mQTLs with non-zero CPP in CAVIAR -->
## Allelic dosage restriction
```{r}

```


## CAVIAR Fine-Mapping
```{r}

```

## TESTING 
Checking if these mQTLs include too many low-frequency snps:
```{r}
maf_table <- fread("placenta_regulatory_landscape/RootStudyConsentSet_phs001717.PlacentalRegulation.v1.p1.c1.HMB-IRB-PUB-COL-MDS/genotype_qc/all_imputed_r2_30_rsid.afreq")
common_SNP <- maf_table[ALT_FREQS >= 0.01 & ALT_FREQS <= 0.99]$ID
hits_common <- na.omit(sig_delahaye[common_SNP, on = "SNP"])
``` 

```{r}
common_snp_rich <- fread("../RICHS_genotyping/phg001102.v1.RICHS.genotype-calls-matrixfmt.MEGA_Consortium.c1.GRU/all_imputed_r2_30_rsid.afreq")[ALT_FREQS >= 0.01 & ALT_FREQS <= 0.99]$ID
nrow(richs_mqtl[SNP %chin% common_snp_rich & FDR < 0.05])
nrow(richs_mqtl[SNP %chin% common_snp_rich & `p-value` < (0.05 / nrow(richs_mqtl))])
length(unique(richs_mqtl[SNP %chin% common_snp_rich & FDR < 0.05]$gene))
length(unique(richs_mqtl[SNP %chin% common_snp_rich & `p-value` < (0.05 / nrow(richs_mqtl))]$gene))
```

```{r}
nrow(robinson_mqtl[FDR < 0.05])
nrow(robinson_mqtl[`p-value` < (0.05 / nrow(robinson_mqtl))])
length(unique(robinson_mqtl[FDR < 0.05]$gene))
length(unique(robinson_mqtl[`p-value` < (0.05 / nrow(robinson_mqtl))]$gene))
```

