---
title: "mQTL Preparation"
output: html_document
---


```{r setup, include=FALSE}
library(tidyverse)
library(data.table)
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "/scratch/st-dennisjk-1/wcasazza/delahaye_QC/")
```

# delahaye

```{r}
delahaye_ancestry <- fread("placenta_regulatory_landscape/RootStudyConsentSet_phs001717.PlacentalRegulation.v1.p1.c1.HMB-IRB-PUB-COL-MDS/genotype_qc/raw_data.geno.maf.mind.sex_check.het_filter.ibd_filter.eigenvec")
delahaye_meta <- fread("placenta_regulatory_landscape/RootStudyConsentSet_phs001717.PlacentalRegulation.v1.p1.c1.HMB-IRB-PUB-COL-MDS/PhenotypeFiles/Placental_Regulation_Phenotypes_clean.txt")
delahaye_meta$IID <- delahaye_meta$SUBJECT_ID
delahaye_meta_methy <- fread("DNAm_w_paired_genotype_metadata.txt")
delahaye_grafpop <- fread("placenta_regulatory_landscape/RootStudyConsentSet_phs001717.PlacentalRegulation.v1.p1.c1.HMB-IRB-PUB-COL-MDS/genotype_qc/raw_data_assigned_ancestry.clst", col.names = c("FID", "IID", "GRAFpop"))
delahaye_meta <- delahaye_meta %>%
  right_join(delahaye_ancestry, by = c("IID" = "V2")) %>%
  right_join(delahaye_meta_methy, by = "SUBJECT_ID") %>%
  left_join(delahaye_grafpop %>% select(-FID), by = c("IID"))
betas_sub <- fread("processed_DNAm_delahaye_quantile_norm.txt")
genotype <- fread("placenta_regulatory_landscape/RootStudyConsentSet_phs001717.PlacentalRegulation.v1.p1.c1.HMB-IRB-PUB-COL-MDS/genotype_qc/raw_data.imputed.r2_30.maf_mismatch.traw")
```

## Match Samples
```{r}
colnames(genotype) <- gsub(".*_", "", colnames(genotype))
delahaye_meta <- delahaye_meta[IID %in% colnames(genotype) & IID %in% colnames(betas_sub)]
gen_vars <- c(colnames(genotype)[1:6], delahaye_meta$IID)
to_delete <- colnames(genotype)[!colnames(genotype) %in% gen_vars]
genotype[, c(to_delete) := NULL]
setcolorder(genotype, gen_vars)
methy_vars <- c("cpg", delahaye_meta$IID)
betas_sub <- betas_sub[, methy_vars, with = FALSE]
```
```{r}
all(colnames(betas_sub)[-c(1)] == delahaye_meta$IID)
all(colnames(betas_sub)[-c(1)] == colnames(genotype)[-c(1:6)])
```
## Compute PCs
```{r}
methy_PC <- prcomp(t(betas_sub[, -c(1)]), center = T, rank. = 10)
```

<!-- ## Write Covars -->

<!-- ```{r} -->
<!-- cat_vars <- model.matrix(~0+plate, data=delahaye_meta %>% mutate(plate= as.factor(plate))) -->
<!-- for(i in 0:20){ -->
<!--   if(i == 0){ -->
<!--     covar<- cbind(pcs,cat_vars,delahaye_meta[,c("V3","V4","V5","age","men")]) -->
<!--   } -->
<!--   else{ -->
<!--     pcs <- methy_PC$x[,1:i] -->
<!--     covar<- cbind(pcs,cat_vars,delahaye_meta[,c("V3","V4","V5","age","men")]) -->
<!--   } -->
<!--   write_delim(t(covar) %>% as.data.frame() %>% rownames_to_column("id"),sprintf("delahaye_data/covariates_%d_methy_PC.txt",i)) -->
<!-- } -->
<!-- ``` -->

## Write data
```{r}
library(minfi)
library(IlluminaHumanMethylation450kanno.ilmn12.hg19)
methy_annot <- as.data.table(getAnnotation(IlluminaHumanMethylation450kanno.ilmn12.hg19))
```
## Split by ancestry
```{r}
table(delahaye_meta$GRAFpop)
ancestry_ids <- list(
  EUR = delahaye_meta[GRAFpop == "EUR"]$IID,
  AFR_AM = delahaye_meta[GRAFpop == "AFR_AM"]$IID,
  LAT_AM = delahaye_meta[grepl("LAT_AM", GRAFpop)]$IID,
  LAT_AM_1 = delahaye_meta[GRAFpop == "LAT_AM_1"]$IID,
  LAT_AM_2_ids = delahaye_meta[GRAFpop == "LAT_AM_2"]$IID
)
```

```{r}
for (ancestry in names(ancestry_ids)) {
  cur_meta <- na.omit(delahaye_meta[match(IID, ancestry_ids[[ancestry]])])
  if (length(unique(cur_meta$batch)) > 1) {
    cat_vars <- model.matrix(~ 0 + batch, data = na.omit(delahaye_meta[match(IID, ancestry_ids[[ancestry]])]) %>% mutate(batch = as.factor(batch)))
    covar <- cbind(methy_PC$x[ancestry_ids[[ancestry]], 1:9], cat_vars, cur_meta[, .(V3, V4, V5, V6, V7, age = Gestational.Age.at.Delivery, men = as.numeric(Gender.x == "Male"))])
    write_delim(t(covar) %>% as.data.frame() %>% rownames_to_column("id"), sprintf("matrix_eqtl_data/covariates_%s.txt", ancestry))
  } else {
    covar <- cbind(methy_PC$x[ancestry_ids[[ancestry]], 1:9], cur_meta[, .(V3, V4, V5, V6, V7, age = Gestational.Age.at.Delivery, men = as.numeric(Gender.x == "Male"))])
    write_delim(t(covar) %>% as.data.frame() %>% rownames_to_column("id"), sprintf("matrix_eqtl_data/covariates_%s.txt", ancestry))
  }
  to_select <- c("cpg", ancestry_ids[[ancestry]])
  fwrite(betas_sub[, ..to_select], sprintf("matrix_eqtl_data/methylation_matrixeQTL_%s.txt", ancestry), sep = "\t", quote = F)
  to_select <- c("SNP", ancestry_ids[[ancestry]])
  fwrite(genotype[, ..to_select], sprintf("matrix_eqtl_data/all_imputed_matrixeQTL_%s.txt", ancestry), sep = "\t", quote = F)
}
```

```{r}
# SNP POS
fwrite(genotype[, .(SNP, CHR = paste0("chr", CHR), POS)], "matrix_eqtl_data/snp_pos.txt", sep = "\t", quote = F)
# SNP POS
fwrite(genotype[CHR == 21, .(SNP, CHR = paste0("chr", CHR), POS)], "matrix_eqtl_data/snp_pos_chr1.txt", sep = "\t", quote = F)
# SNPs
geno_vars <- colnames(genotype)[c(1, 3:6)]
genotype[, c(geno_vars) := NULL]
fwrite(genotype, "matrix_eqtl_data/all_imputed_matrixeQTL.txt", sep = "\t", quote = F)
# Methy POS
fwrite(methy_annot[Name %in% betas_sub$cpg, .(geneid = Name, chr, s1 = pos, s2 = pos)], "matrix_eqtl_data/probe_pos.txt", sep = "\t", quote = F)
# methy
fwrite(betas_sub, "matrix_eqtl_data/methylation_matrixeQTL.txt", sep = "\t", quote = F)


# SNPs
snp_pos <- fread("matrix_eqtl_data/snp_pos.txt")
fwrite(genotype[SNP %in% snp_pos[CHR == "chr21"]$SNP], "matrix_eqtl_data/all_imputed_matrixeQTL_chr1.txt", sep = "\t", quote = F)
# Methy POS
fwrite(methy_annot[Name %in% betas_sub$cpg & chr == "chr21", .(geneid = Name, chr, s1 = pos, s2 = pos)], "matrix_eqtl_data/probe_pos_chr1.txt", sep = "\t", quote = F)
chr1_cpg <- methy_annot[Name %in% betas_sub$cpg & chr == "chr21", ]$Name
# methy
fwrite(betas_sub[cpg %in% chr1_cpg], "matrix_eqtl_data/methylation_matrixeQTL_chr1.txt", sep = "\t", quote = F)
```


```{r}
fmt <- "matrix_eqtl_data/mQTL_covar_%d_methy_PC.txt"
batch_matrix <- model.matrix(~batch, data = delahaye_meta) %>% as.data.frame()
colnames(batch_matrix) <- gsub(" ", "_", colnames(batch_matrix))
colSums(batch_matrix)
for (i in 0:20) {
  if (i == 1) {
    tmp <- t(cbind(PC1 = methy_PC$x[, 1:i], batch = batch_matrix[, -c(1)], delahaye_meta %>% as.data.frame() %>%
      dplyr::select(
        Gestational.Age.at.Delivery,
        V3,
        V4,
        V5,
        V6,
        V7,
        Gender.x
      )) %>%
      mutate(Sex = as.numeric(Gender.x == "Male")) %>%
      dplyr::select(-Gender.x)) %>%
      as.data.frame()
  } else if (i == 0) {
    tmp <- t(cbind(batch = batch_matrix[, -c(1)], delahaye_meta %>% as.data.frame() %>%
      dplyr::select(
        Gestational.Age.at.Delivery,
        V3,
        V4,
        V5,
        V6,
        V7,
        Gender.x
      )) %>%
      mutate(Sex = as.numeric(Gender.x == "Male")) %>%
      dplyr::select(-Gender.x)) %>%
      as.data.frame()
  } else {
    tmp <- t(cbind(methy_PC$x[, 1:i],
      batch = batch_matrix[, -c(1)],
      delahaye_meta %>% as.data.frame() %>%
        dplyr::select(
          Gestational.Age.at.Delivery,
          V3,
          V4,
          V5,
          V6,
          V7,
          Gender.x
        )
    ) %>%
      mutate(Sex = as.numeric(Gender.x == "Male")) %>%
      dplyr::select(-Gender.x)) %>%
      as.data.frame()
  }
  colnames(tmp) <- delahaye_meta$IID
  write_delim(
    tmp %>% rownames_to_column(var = "id"),
    path = sprintf(fmt, i)
  )
}
```

## PC checking

```{r}
fmt <- "matrix_eqtl_data/cis_all_impute_mQTL_results_%d_methy_PC_chr21.txt"
chr21_res <- lapply(
  0:20,
  function(i) {
    tmp <- fread(sprintf(fmt, i))
    nrow(tmp[`p-value` < (0.05 / .N)])
  }
)
```
```{r}
to_plot <- data.frame(PC = 0:20, hits = unlist(chr21_res))
ggplot(to_plot, aes(PC, hits)) +
  geom_point() +
  geom_smooth()
```
